packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation_objc/FLTCam.m

```diff
@@ -846,40 +846,49 @@ - (CVPixelBufferRef)copyPixelBuffer {
 - (void)startVideoRecordingWithCompletion:(void (^)(FlutterError *_Nullable))completion
                     messengerForStreaming:(nullable NSObject<FlutterBinaryMessenger> *)messenger {
   if (!_isRecording) {
-    if (messenger != nil) {
-      [self startImageStreamWithMessenger:messenger];
-    }
+    // Main logic to setup the video recording in a block to reuse in the completion block.
+    void (^videoRecordingSetupWithCompletion)(void) = ^void(void) {
```

Comment:
Please make this a helper method rather than a block; having this here as a block makes the flow much harder to follow.